### PR TITLE
QF-918 Upgrade to Azure Identity

### PR DESCRIPTION
**Pull request by SawNawDfE**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeFeedback.Api/AppStart/NServiceBusRegistration.cs [hunk 1]
```diff
@@ -67,7 +68,8 @@ private static DbConnection CreateSqlConnection(IConfiguration configuration, Ap
                 return new SqlConnection
                 {
                     ConnectionString = appSettings.DbConnectionString,
-                    AccessToken = new AzureServiceTokenProvider().GetAccessTokenAsync("https://database.windows.net/").Result
+                    AccessToken = new ChainedTokenCredential().GetTokenAsync(
+                        new TokenRequestContext(new string[] { "https://database.windows.net/" })).Result.Token
```
Comment (SawNawDfE) [inline src/SFA.DAS.ApprenticeFeedback.Api/AppStart/NServiceBusRegistration.cs:72]: I'm not sure about this one. Here I've used [this PR](https://github.com/SkillsFundingAgency/das-apim-developer-api/pull/3/files#diff-8a6c98ca86a8d24b1ed7d3f79b38737bbdfacbb14091d4dbfbb47244fd76e5e9) as a guide and used `ChainedTokenCredential`, but the [official migration guide](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/app-auth-migration?view=azure-dotnet) suggests using `DefaultAzureCredential`, and makes no mention of the `ChainedTokenCredential` used in the other PR.

Which is the correct way?

Comment (dashton82) [reply src/SFA.DAS.ApprenticeFeedback.Api/AppStart/NServiceBusRegistration.cs:72]: What you've done will work - however the _correct_ way would be to just use the DefaultAzureCredential and allowing the details of the connection string to determine how it obtains the access token.

Comment (SawNawDfE) [reply src/SFA.DAS.ApprenticeFeedback.Api/AppStart/NServiceBusRegistration.cs:72]: I've changed it to use DefaultAzureCredential, does it look alright?

**Comment by SawNawDfE** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68)
> 
> [Bug](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=SkillsFundingAgency_das-apprentice-feedback-api&pullRequest=68&metric=new_duplicated_lines_density&view=list)

@ChrisJWoodcock Hurrah! Code smell eliminated

**Review by ChrisJWoodcock — APPROVED** [review]
It would interesting to see if this works without the singletons added in AddDatabaseRegistrations.cs you could try testing it without on AT

**Pull request closed without merging**